### ISSUE TITLE
Hint component

### ISSUE THREAD

**Issue by stacymcauliffe** [patternfly/patternfly-design-kit#177]
Create a PF component for the hint pattern as documented in the [OpenShift design wiki](http://openshift.github.io/openshift-origin-design/web-console/future-openshift/contextual-help/contextual-help) and [draft PF design guidelines](https://docs.google.com/document/d/1v-_QA2DuDzKk73OJG5TZDTdQwnSQBec6J-Q-yKPRKKs/edit?usp=sharing) for embedded assistance.

Hint pattern should meeting the following requirements:
Hint area is shown in context of where it applies
- This may be above the page title for cases where the hints apply to the entire page of the application.
- This may be below a specific tab or area on a page for cases where the hints only apply to a portion.

Hint area can be different shapes and sizes to fit a specific context (for example, could be a horizontal rectangle - like an inline alert - if applies to an entire page or page area, or could take a vertical rectangle to fit in white space on the right side of a modal or page)

Hint area is highlighted in blue
Hint must have a title (h3 style)
Hint must have some body text (body-copy style)
There can optionally be external links within the body text
There can optionally be a primary action below the body text
Title, body text, and primary action are left aligned
For cases without a primary action, there should be an X in the top right corner to dismiss the message.

**Colors**
Font: pf-blue-600
Background: pf-blue-50
Primary Action: Standard PatternFly button colors

**Comment by gdoyle1** [comment]
(symbols) Will need to add the active/selected card symbol which we'll have to work in tangent with the #173. Also need a "dominant" card style symbol for this.
(template) Full screen demo of this in action

**Comment by maryshak1996** [comment]
@gdoyle1 the symbol library updates are all set! once this is merged, I'll create the template :) 
[hint-symbol-addition.sketch.zip](https://github.com/patternfly/patternfly-design-kit/files/4345084/hint-symbol-addition.sketch.zip)

**Comment by gdoyle1** [comment]
Merged the symbols! @maryshak1996

**Comment by maryshak1996** [comment]
[template_with-hint.sketch.zip](https://github.com/patternfly/patternfly-design-kit/files/4345805/template_with-hint.sketch.zip)
here's the template! should be all set to close now :)

**Comment by ajacobs21e** [comment]
@maryshak1996 or @mcarrano - is only the dominant version recommended for the 'pinned to header' option? I have a use case where I want something pinned to the top of the content area but it will include a link. I think that means I should use the hint instead of the banner. But ideally i'd like it full width & pinned to the top. Can I do that with the default hint? Thanks!

**Comment by mcarrano** [comment]
@maryshak1996 I don't remember our conversation about this. Was there a reason that pinning was only supported for a dominant hint or is it just being read that way from the documentation?

**Comment by ajacobs21e** [comment]
@mcarrano I did try the default hint pinned to header. I shared it with Abi for content help and she also gave me this design feedback:

> I actually didn’t initially see the hint on 4 and 5 when I was flipping thru. And the skinny blue banner on 6 and 7 is hard to read, but it does jump out a bit more. Maybe a thicker blue banner?

I agree with her that the default hint isn't very obvious when pinned to the edges. I also agree that the banner is very small and hard to read. I can share my wires with you and Mary tomorrow if we want to talk through this and iron out the use cases for the documentation. Thanks!

**Comment by mcarrano** [comment]
@ajacobs21e The Banner is already being implemented and therefore I'm inclined to not make further changes at this point. It is only intended for short informational messages and not to be super prominent. The Hint is in the queue for May. 

Yes, it would be great if you can share your wires? We could pull Hint back if needed, but I will say that we reviewed multiple iterations with stakeholders during our weekly meetings before arriving at the current design. So if we need to make further changes will want to first understand why this does not work.

**Comment by mcarrano** [comment]
@gdoyle1 @maryshak1996 Can we hold on this for the moment? After discussing alert changes with @mceledonia and @kybaker (see https://github.com/patternfly/patternfly-design/issues/804) we decided to revisit Hint appearance. I have opened https://github.com/patternfly/patternfly-design/issues/837 to track that work.

@ajacobs21e if you have input regarding the Hint feel free to comment there.

**Comment by gdoyle1** [comment]
> @gdoyle1 @maryshak1996 Can we hold on this for the moment? After discussing alert changes with @mceledonia and @kybaker (see [patternfly/patternfly-design#804](https://github.com/patternfly/patternfly-design/issues/804)) we decided to revisit Hint appearance. I have opened [patternfly/patternfly-design#837](https://github.com/patternfly/patternfly-design/issues/837) to track that work.
> 
> @ajacobs21e if you have input regarding the Hint feel free to comment there.

It has already been merged into the library so we'll have to keep it there for now. @maryshak1996 and I will talk to see if we can delete it without it breaking existing symbols users are using next release!

**Comment by mcarrano** [comment]
@gdoyle1 If you've already added it, I would not jump through extra hoops to remove it. We can always make updates if we decide to rework the design. This issue should probably be closed then.

**Comment by gdoyle1** [comment]
Ok going to close this and if we make more updates to the design then we will update in the library based on that story that opens up!

**Comment by maryshak1996** [comment]
[Final design doc](https://docs.google.com/document/d/1qAYYv-VPuOst1o-BFx0R7Wgh3jt6X9ATmV4WQU5hXcw/edit?usp=sharing)
---
@mcarrano Designs have been updated based on the recent conversations around the default style. Before:
![image](https://user-images.githubusercontent.com/19824574/78939637-88f02c80-7a82-11ea-968e-55037594b4e1.png)
After
![image](https://user-images.githubusercontent.com/19824574/78939652-8c83b380-7a82-11ea-9707-e5056e291a1e.png)

**Comment by maryshak1996** [comment]
Re-opening here, since we have changed the default style as show in comment above. We need to add this to the Sketch libarary.
@gdoyle1 I can crank this out real quick :)

**Comment by maryshak1996** [comment]
just pushed the update (gave the background a temporary name for the appearance, and will reopen + change the name once it's implemented)!

**Comment by gdoyle1** [comment]
Thanks @maryshak1996 !
